src/uu/cksum/src/cksum.rs

```diff
@@ -116,52 +117,78 @@ where
         };
         // The BSD checksum output is 5 digit integer
         let bsd_width = 5;
-        match (options.algo_name, not_file) {
-            (ALGORITHM_OPTIONS_SYSV, true) => println!(
-                "{} {}",
-                sum.parse::<u16>().unwrap(),
-                div_ceil(sz, options.output_bits)
+        let (before_filename, print_filename, after_filename) = match (options.algo_name, not_file)
+        {
+            (ALGORITHM_OPTIONS_SYSV, true) => (
+                format!(
+                    "{} {}",
+                    sum.parse::<u16>().unwrap(),
+                    div_ceil(sz, options.output_bits)
+                ),
+                false,
+                String::new(),
             ),
-            (ALGORITHM_OPTIONS_SYSV, false) => println!(
-                "{} {} {}",
-                sum.parse::<u16>().unwrap(),
-                div_ceil(sz, options.output_bits),
-                filename.display()
+            (ALGORITHM_OPTIONS_SYSV, false) => (
+                format!(
+                    "{} {} ",
+                    sum.parse::<u16>().unwrap(),
+                    div_ceil(sz, options.output_bits)
+                ),
+                true,
+                String::new(),
```

Comment:
The difference between the two arms is difficult to see (a space at the end of the format string), though currently I don't see how this could be improved.

Reply:
Maybe like this ?
```rust
(ALGORITHM_OPTIONS_SYSV, not_file) => {
    let mut s = format!(
        "{} {} ",
        sum.parse::<u16>().unwrap(),
        div_ceil(sz, options.output_bits)
    );

    // If file, remove the last space.
    if !not_file {
        // Assume all chars are ASCII.
        s.truncate(s.len() - 1);
    }

    (s, false, String::new())
}
```

Reply:
I solved it by removing `notfile` from the `match` structure entirely, and instead pushing it into the `format` for `before_filename`.